src/Ocelot/Configuration/Creator/SecurityOptionsCreator.cs

```diff
@@ -35,5 +27,21 @@ public SecurityOptions Create(FileSecurityOptions securityOptions)
 
             return new SecurityOptions(ipAllowedList, ipBlockedList);
         }
+
+        private static List<string> SetIpAddressList(List<string> ipValueList)
+        {
+            var ipList = new List<string>();
+
+            foreach (var ipValue in ipValueList)
+            {
+                if (IPAddressRange.TryParse(ipValue, out var ipAddressRange))
+                {
+                    var ips = ipAddressRange.Select<IPAddress, string>(x => x.ToString());
+                    ipList.AddRange(ips);
+                }
+            }
+
+            return ipList;
+        }
```

Comment:
```suggestion
       private static IReadOnlyCollection<string> SetIpAddressList(IList<string> ipValueList)
{
    return ipValueList
        .Where(ipValue => IPAddressRange.TryParse(ipValue, out _))
        .SelectMany(ipValue => IPAddressRange.Parse(ipValue).Select(ip => ip.ToString()))
        .ToArray();
}
```

Reply:
@RaynaldM I've to modify `.Select(ip => ip.ToString())` into `.Select<IPAddress, string>(ip => ip.ToString())` otherwise the code doesn't build.  I've to modify also the `SecurityOptions` class due to the constructor parameters actually set to `List<string>`

Reply:
You may modify the `SecurityOptions` constructors to generalize with more abstract types such as `IList<string>` or `IEnumerable<string>` if necessary.

Reply:
@raman-m yes, my comment is strictly referred to _Commit suggestion action_ because the project won't build 😊

Reply:
commit reverted and splitted into 3 different ones.